### DISCUSSION TITLE
Need help with Sorting in PnP Results Page

### DISCUSSION THREAD

**Discussion by RamBalani** [microsoft-search/pnp-modern-search#2045] [General]
I can’t seem to find a way to sort inside PnP my List items by the assigned List ID numerically, is there some trick /tip to do that inside PnP?

I made the ID column Sort-enabled within Manage Column but sorting scheme is cock-eyed and not as it is in the source list.

To explain more:

In the source list-sorts like this below. Note 2nd column below ID – **sorts 1,2,3 in the source Lis**t. See Attachment 1 Screen shot.

In the PnP contained list -see below. Note in last column **![Sorting Screenshot in Source List ID](https://user-images.githubusercontent.com/37209670/165944943-b27774ca-2ec7-41c5-a9d6-cb676cd0e52a.png)** - See attachment 2 screenshot.
![Sorting Screenshot in PnP ResultsPage is Sorting String not Integer ID](https://user-images.githubusercontent.com/37209670/165945039-6db26f9e-5235-4cfe-8b70-1c52810a8878.png)

**Comment by wobba** [comment]
`ListItemID` is not sortable. You can either sort on created date instead, as that should reflect the list id order. Or you can map `ows_ListItemID` to e.g `RefinableInt01` and sort on that.

**Comment by RamBalani** [comment]
![Screenshot -PNP Managed Column-Last ColumnNot Showing up](https://user-images.githubusercontent.com/37209670/166009668-da5b007c-4925-4b01-8494-a8e6bc82a8bc.jpg)
Thanks for the feedback, we did no know ListItemID is not sortable. That said-nice to know we're picking up some know-how on our own though still no 'cigar' so to speak. 
We have, on our own prior to posting in GitHub, already mapped **ows_ListItemID crawled  property to RefinatbleInt00 mapped property** as you suggested here  since created date will likely not do, i.e list items were created all at one time via Export from an Excel worksheet into SP online site.

We have also  attempted to use RefinableInt00 within PnP as managed column that is sortable-no luck, i.e. RefinatbeInt00 does not show up! 

What are we doing wrong or still not doing? Is an 'edit sort' needed? A slot layout? A handlebar for this to show up? We're lost ! (* though we know we are close*)
![Screenshot Mapping ![Screenshot -Selected Properties Show RefinableInt00-01 etc](https://user-images.githubusercontent.com/37209670/166009657-d3470698-4e7e-4206-99f3-18ad9b3015a8.jpg) to RefinableInt00 -Shared Github](https://user-images.githubusercontent.com/37209670/166009625-4ab8b617-037d-4718-8cf2-fa9712b6f357.jpg)

**Comment by RamBalani** [comment]
Here's what we have on the Managed Column to be clear, sharing more screenshots.
Attachment shows Managed Columns  set-up for RefinableInt00.
Still no 'cigar' for some reason -last column on PnP Results does not show the RefinableInt00 data made sortable.
![Screenshot -Manage Column 6-Last Column is RefinableInt00](https://user-images.githubusercontent.com/37209670/166024268-56f638d0-0009-4ce6-b402-a15c6179e79f.jpg)

![Screenshot -ManagedColumn Handlebar for RefinatbleInt00](https://user-images.githubusercontent.com/37209670/166025027-9480fc58-ec5d-4124-96db-e989cdba382a.jpg)

**Reply by wobba** [reply]
(If you view the managed properties each one will say if it's sortable, refinable etc (and refinable ones are also sortable). If a prop don't show up in a picker, then type it manually into the combobox and click enter.)

After you map a cp to a mp, you need to re-index the data in order to populate the mp. And you can use e.g. SharePoint Search Query Tool to inspect values of properties to ensure data is there before setting up the web parts. Or add the property to the list of properties to return and use the debug display template to verify data.

**Comment by RamBalani** [comment]
No issues with RefinableInt00 showing up on the Managed Column props--have verified it many times-it's 'living' there already.
Now this maybe the key, i.e. after mapping ows_ListItemID to RefinableInt00 which we use in PnP- the site was not mapped though I did add data on the List thinking that would be sufficient to prime an index on the list level.
Just re-index the site, giving it some time to do its thing-will verify and circle back.
Not familiar with Sharepoint Search Tool though I remember coming across that in your book !-Thanks for bringing up. Will prole debug display as well-another new thing to us mere mortals 🚀.
Many thanks for the tips. Back soon with news.

**Reply by wobba** [reply]
Any new item added should have the I'd mapped and available. And if you follow the "sort settings" section at https://microsoft-search.github.io/pnp-modern-search/usage/search-results/data-sources/ you should be able to make it work once the mp has data in it.

**Comment by RamBalani** [comment]
Now working! Thanks.

I re-indexed at the site and list source  both - now working with sort on List ID column using RefinableInt00 on PnP web part screen. Re-indexing did the trick.

💡However-one list item ( # 11) was missing  on the PnP Web part screen , i.e.  not appearing in sequence numerically expected - puzzling 🍳 for some reason I could not discern.
.
So had to go to the list source--there --- List item # 11  complete with ID shows as a visible List item with data.
I then  verified by PnP Search box keyword that indeed List item 11 was in the List data imported into PnP web part -but missing an ID column value, i.e. ID of item $ 11 = blank?

Not knowing what the heck was happening- decided to open up list item source 11 on 'edit' mode,  did nothing-just re-saved.
PnP web part results then worked,  this time showing List item # 11 as numerically in ID column sort order expected sequence.. Great example of the MS SP Online  'vodoo' beyond my pay grade to analyze.🚀

**Comment by RamBalani** [comment]
### Speaking of 'vodoo' here' another one for you.
Though Refinableint00 defined in 'edit sort order' has remedied on entry to PnP web part the list items are sorted in **ID numeric ascending sequence** as in the list source, tweaking the click sort arrow next to the screen last column reverts to sorting by string. See below.

**One for the books.**

1-1st image is on entry into PnP Web part  -right sequence-sorts by ID number in ascending  order numerically.
![Screenshot -on Entry REV3-ISO-13485-PnP-Use Web Part-Sorts correct](https://user-images.githubusercontent.com/37209670/166107287-929b4dc2-3129-4766-8965-2e599aaa4bd8.jpg)

2-There are 15 list items, 1 thru 15 by ID in numeric ascending ID sequence number on the PnP web part screen per page.
 I tweak the last column to **descending order -** mucked up!

List page on PnP screen contains 15 list items (1 - 15 numerically ID'ed) -  tweaking column sort order to descending sort I expect to go 15,14,13,12,11,10,9,8...1 if sorted numerically--it does not, instead it shows cock-eyed again reverting to sorting as strings.
 
See attachments.

1-on PnP Web part screen entry-**CORRECT !**
![Screenshot -on Entry REV3-ISO-13485-PnP-Use Web Part-Sorts correct](https://user-images.githubusercontent.com/37209670/166107643-81c53719-1683-4e79-a2de-eae7276438a4.jpg)

2-When tweaking Sort Column (Last col on screen down arrow to descending) - mucked up! Returns to sorting by string though Sortable drop box on managed column is set to sort also on RefinableInt00!
![Screenshot -on Entry REV3-ISO-13485-PnP-Use Web Part-SortClick-MuckedupShow5 ](https://user-images.githubusercontent.com/37209670/166107705-c4d314e7-d53c-46e6-9d9a-0d61c5be9909.jpg)
-------
![Screenshot -on Entry REV3-ISO-13485-PnP-Use Web Part-SortClick-MuckedupShow15](https://user-images.githubusercontent.com/37209670/166107715-11b0e324-72a2-428d-b1cd-00abf754d952.jpg)

Not sure if a bug or just an un-documented PnP Search SPFx Web part feature working correctly by design and default.  Which is it ?

**Reply by wobba** [reply]
Please log this as a separate issue and not voodoo. Just a scenario not thought of when sorting.

**Comment by RamBalani** [comment]
No problems-will do. DO i need to link the new issue to this Discussion Post ? We have not done that before with Github.
That said--so this is not a bug but a default behavior inherited or designed into PnP  then -correct?, i.e. PnP sortable column assumes  default string  based  sort inside PnP Search Web part ?
'Vodoo' used  here was meant to imply  a 'mystery'- nothing more beyond that, no offense intended.

**Reply by wobba** [reply]
You can link of you want. And it's likely a missed scenario.

**Comment by RamBalani** [comment]
Okay thanks.

**Comment by RamBalani** [comment]
Posted as new Open issue just now ---------------------------------------------------------------------------------------- ---------------
"PnP Search Web Part -Sortable Column Setting Defaults to String Sort Regardless 'Edit Sort Order' #2053" , ie. our first issue post -hope was done right.
Thanks.
